test/test_images/readiness/readiness.go

```diff
@@ -87,7 +76,22 @@ func main() {
 		time.Sleep(delay)
 	}
 
-	http.ListenAndServe(":8080", nil)
+	mainServer := http.NewServeMux()
+	mainServer.HandleFunc("/", handleMain)
+	mainServer.HandleFunc("/start-failing", handleStartFailing)
+
+	probeServer := http.NewServeMux()
+	probeServer.HandleFunc("/", handleHealthz)
+
+	go func() {
+		http.ListenAndServe(":"+strconv.Itoa(healthcheckPort), probeServer)
+	}()
+
+	go func() {
+		http.ListenAndServe(":8080", mainServer)
+	}()
+
+	<-finish
```

Comment:
You can drop the use of the channel and just call `http.ListenAndServe` to block

Reply:
I think the go func is required cause you have two ListenAndServe, which doesn't work on the same thread

Reply:
If you invoke one `ListenAndServe` in a go routine and the other on the main go routine you don't need the `finish` channel